src/Cli.elm

```diff
@@ -93,6 +107,41 @@ program =
             )
 
 
+effectTypesValidation : Maybe String -> Result String (List EffectType)
+effectTypesValidation str =
+    case str of
+        Nothing ->
+            Ok [ Cmd, Task ]
+
+        Just v ->
+            v
+                |> String.split ","
+                |> List.map String.trim
+                |> Result.Extra.combineMap effectTypeValidation
+
+
+effectTypeValidation : String -> Result String EffectType
+effectTypeValidation effectType =
+    case effectType of
+        "cmd" ->
+            Ok Cmd
+
+        "cmdrisky" ->
+            Ok CmdRisky
+
+        "task" ->
+            Ok Task
+
+        "taskrisky" ->
+            Ok TaskRisky
+
+        "backendtask" ->
+            Ok BackendTask
+
+        _ ->
+            Err <| "Unexpected effect type: " ++ effectType
```

Comment:
This is reminding me of the horrors of a CLI at Square written in Go that didn't handle bad values

Reply:
Oh dear.
I really appreciate that Elm forces us to handle all cases